network.go

```diff
@@ -138,6 +170,61 @@ func (n *VirNetwork) GetXMLDesc(flags uint32) (string, error) {
 	return xml, nil
 }
 
+func (n *VirNetwork) UpdateXMLDesc(xmldesc string, command, section int) error {
+	xmldescC := C.CString(xmldesc)
+	result := C.virNetworkUpdate(n.ptr, C.uint(command), C.uint(section), C.int(-1), xmldescC, C.uint(C.VIR_NETWORK_UPDATE_AFFECT_CURRENT))
+	C.free(unsafe.Pointer(xmldescC))
+	if result == -1 {
+		return GetLastError()
+	}
+	return nil
+}
+
+type NetworkHost struct {
+	XMLName xml.Name `xml:"host"`
+	Mac     string   `xml:"mac,attr,omitempty"`
+	Name    string   `xml:"name,attr,omitempty"`
+	IP      string   `xml:"ip,attr,omitempty"`
+}
+
+func getHostXMLDesc(ip, mac, name string) (string, error) {
+	host := NetworkHost{
+		Mac:  mac,
+		Name: name,
+		IP:   ip,
+	}
+
+	b, err := xml.Marshal(host)
+	if err != nil {
+		var virErr VirError
+		virErr.Code = VIR_ERR_XML_ERROR
+		virErr.Message = fmt.Sprintf("Invalid host entry definition: %s", err)
+
+		return "", virErr
+	}
+	return string(b), nil
+}
+
+// Adds a new static host to the network
+func (n *VirNetwork) AddHost(ip, mac, name string) error {
+	hostXml, err := getHostXMLDesc(ip, mac, name)
+	if err != nil {
+		return err
+	}
+	return n.UpdateXMLDesc(hostXml,
+		C.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST, C.VIR_NETWORK_SECTION_IP_DHCP_HOST)
+}
+
+// Removes a static host from the network
+func (n *VirNetwork) RemoveHost(ip, mac, name string) error {
+	hostXml, err := getHostXMLDesc(ip, mac, name)
+	if err != nil {
+		return err
+	}
+	return n.UpdateXMLDesc(hostXml,
+		C.VIR_NETWORK_UPDATE_COMMAND_DELETE, C.VIR_NETWORK_SECTION_IP_DHCP_HOST)
+}
+
 func (n *VirNetwork) Undefine() error {
```

Comment:
I have the same problem than with #1 

The utility methods should go into terraform provider. Upstream will not let me upstream public APIs that are not part of libvirt API.